mongoengine/context_managers.py

```diff
@@ -220,7 +220,7 @@ def __repr__(self):
     def _get_count(self):
         """ Get the number of queries. """
         ignore_query = {"ns": {"$ne": "%s.system.indexes" % self.db.name}}
-        count = self.db.system.profile.find(ignore_query).count() - self.counter
+        count = self.db.system.profile.estimated_document_count(query=ignore_query) - self.counter
```

Comment:
This doesn't seem correct. If we're filtering the documents in the collection by an `ignore_query`, shouldn't we use `count_documents` instead?

Reply:
The approach here was to aim for minimal change with `estimated_document_count` while targeting `pymongo>=4.2,<5` since the underlying mechanism there is identical to the one in `pymongo>=3.11`.

But yes, I don't see how it could be an issue in this instance to stick with `count_documents` - I can get that in place with the backwards compatibility for `pymongo>=3.11` we spoke about earlier.